lib/basevm.py

```diff
@@ -13,14 +13,25 @@ class BaseVM:
     """ Base class for VM and Snapshot. """
 
     xe_prefix = "vm"
+    uuid: str
 
-    def __init__(self, uuid, host: 'lib.host.Host'):
+    def __init__(self, uuid: str, host: 'lib.host.Host'):
         logging.info("New %s: %s", type(self).__name__, uuid)
         self.uuid = uuid
         self.host = host
 
+    @overload
+    def param_get(self, param_name: str, key: Optional[str] = ...,
+                  accept_unknown_key: Literal[False] = ...) -> str:
+        ...
+
+    @overload
+    def param_get(self, param_name: str, key: Optional[str] = ...,
+                  accept_unknown_key: bool = ...) -> Optional[str]:
```

Comment:
shouldn't that rather be `accept_unknown_key: Literal[True]`?

Reply:
No, it doesn't happen because `_param_get` has all of the overload needed. But having `Literal[False]` and `Literal[True]` is not enough to be recognized as `bool`.
https://mypy.readthedocs.io/en/stable/literal_types.html#literal-types

Reply:
Oh.  But what I understand of this is that in the general case we need 3 overloads, 2 for the literal values, and one for bool variables, when we cannot tell.  It happens that in our case the union of the return types is the return type for the `Literal[False]` case, but then it would at least warrant a comment explaining that (or just provide the 3 overloads).

Reply:
Indeed, that was I did at first: https://github.com/xcp-ng/xcp-ng-tests/pull/292#discussion_r2027746839 ^^